http/http_client.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+type httpClient struct {
+	client  *http.Client
+	baseURL *url.URL
+	txValue string
+}
+
+func newHttpClient(baseURL *url.URL) *httpClient {
+	client := httpClient{
+		client:  http.DefaultClient,
+		baseURL: baseURL,
+	}
+	return &client
+}
+
+func (c *httpClient) withTxn(value uint64) *httpClient {
+	return &httpClient{
+		client:  c.client,
+		baseURL: c.baseURL,
+		txValue: fmt.Sprintf("%d", value),
+	}
+}
+
+func (c *httpClient) setDefaultHeaders(req *http.Request) {
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Content-Type", "application/json")
+
+	if c.txValue != "" {
+		req.Header.Set(TX_HEADER_NAME, c.txValue)
+	}
+}
+
+func (c *httpClient) request(req *http.Request) ([]byte, error) {
+	c.setDefaultHeaders(req)
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close() //nolint:errcheck
```

Comment:
todo: I can't spot the reason as to why ignoring the error here is okay/desirable - can you please document this in the code with a comment if you feel like we want to do this, or handle the error.

Reply:
Added